core/src/main/scala/org/apache/spark/scheduler/Stage.scala

```diff
@@ -118,7 +119,7 @@ private[scheduler] abstract class Stage(
    */
   private[scheduler] def failedOnFetchAndShouldAbort(stageAttemptId: Int): Boolean = {
     fetchFailedAttemptIds.add(stageAttemptId)
-    fetchFailedAttemptIds.size >= Stage.MAX_CONSECUTIVE_FETCH_FAILURES
+    fetchFailedAttemptIds.size >= maxConsecutiveFetchFailure
```

Comment:
What about instead cleaning this code path up a bit (this isn't the fault of your PR, but your PR could help make things better here!) by:

(1) eliminate this method
(2) make the fetchFailedAttemptIds variable non-private
(3) In the DAGScheduler on line 1268 (right after the else), add "stage.fetchFailedAttemptIds.add(task.stageAttemptId"
(4) change the call to this method to instead directly check "(stage.fetchFailedAttemptIds.size >= maxConsec..."

IMO as is, this function is hurting understandability as-is.  This change would also allow you to eliminate maxConsecutiveFetchFailure from the Stage object altogether.